supreme_court_predictions/data/statistics/DataCleaner.py

```diff
@@ -0,0 +1,225 @@
+
+        # Get local directory
+        cwd = os.getcwd()
+        self.LOCAL_PATH = cwd.replace("\\", "/")
+        self.LOCAL_PATH = self.LOCAL_PATH.replace(
+            "data/statistics", "data/convokit"
+        )
+        print(f"Working in {self.LOCAL_PATH}")
+
+        # Set output path
+        self.OUTPUT_PATH = self.LOCAL_PATH.replace("convokit", "clean_convokit")
+        print(f"Data will be saved to {self.OUTPUT_PATH}")
+
+        if not downloaded_corpus:
+            print("Corpus not present, downloading...")
+            self.get_data()
+        else:
+            print("Corpus already downloaded")
+
+    def get_data(self):
+        """
+        Loads and outputs the Supreme Court Corpus data.
+        """
+
+        print("Loading Supreme Court Corpus Data...")
+        corpus = Corpus(filename=download("supreme-corpus"))
+        corpus.dump("supreme_corpus", base_path=self.LOCAL_PATH)
+
+    # Begin reading data
+    def load_data(self, file_name):
+        """
+        Opens the data and returns it as a dictionary.
+
+        :param file_name: The name of the file to open
+        :return: The data as a dictionary
+        """
+
+        path = self.LOCAL_PATH + f"/supreme_corpus/{file_name}"
+        if "jsonl" in file_name:
+            data = []
+            with open(path) as json_file:
+                json_list = list(json_file)
+
+            for json_str in json_list:
+                clean_json = json.loads(json_str)
+                data.append(clean_json)
+        else:
+            with open(path) as file:
+                data = json.load(file)
+        return data
+
+    def speakers_to_df(self, speakers_dict):
+        """
+        Converts the speakers dictionary to a pandas dataframe.
+
+        :param speakers_dict: The speaker's dictionary
+        :return: The speakers dataframe
+        """
+
+        dict_list = []
+        for speaker_key in list(speakers_dict.keys()):
+            speaker_data = speakers_dict[speaker_key]["meta"]
+            speaker_data["speaker_key"] = speaker_key
+            dict_list.append(speaker_data)
+
+        df = pd.DataFrame(dict_list)
+        df.rename(
+            columns={
+                "name": "speaker_name",
+                "type": "speaker_type",
+                "role": "speaker_role",
+            },
+            inplace=True,
+        )
+        return df
+
+    def get_conversation_dfs(self, conversations_dict):
+        """
+        Converts the conversations dictionary to several pandas dataframes.
+
+        :param conversations_dict: The conversations' dictionary
+        :return: The conversations dataframe, advocates dataframe,
+                and voters dataframe
+        """
+        metadata_list = []
+        advocates_list = []
+        voters_list = []
+
+        for conversation_id in list(conversations_dict.keys()):
+            clean_dict = {}
+            conversation_data = conversations_dict[conversation_id]["meta"]
```

Comment:
🔬 Since we are using these keys throughout this whole class, I think there'd be some value to creating constants at the file level for them.